src/Components/Login.tsx

```diff
@@ -1,121 +1,107 @@
+import {  useNavigate, Link } from 'react-router-dom';
+import useAuth from '../Hooks/useAuth';
+import { Typography } from '@mui/material';
+import useServiceLayer from '../serviceLayer/servicelayer';
 
 export default function Login() {
-  const [user, setUser] = useState<User>(new User("", ""));
-  const [valid, SetValid] = useState<boolean>(true);
-  const [showSnackBar, SetShowSnackBar] = useState(false);
-  const [err, setErr] = useState("");
-  const navigate = useNavigate();
-  const [signIn, signOut] = useAuth();
-  function isValid(user: User) {
-    if (user?.password?.length! >= 8 && user!.name!.trim().length > 1) {
-      return true;
-    }
-    return false;
-  }
-  async function validateUser() {
-    try {
-      var res = await axios.post(
-        `${config.API_KEY}/User/checkuser`,
-        {
-          userName: user.name,
-          userPassword: user.password,
-        }
-      );
-      if (res.status == 200 && res.data["isValid"]) {
-        SetShowSnackBar(true);
-        SetValid(true);
-        signIn(user);
-        navigate("/");
-      } else {
-        SetShowSnackBar(true);
-        SetValid(false);
-        setErr("invalid");
-      }
-      setTimeout(() => {
-        SetShowSnackBar(false);
-      }, 1000);
-    } catch (e) {
-      SetShowSnackBar(true);
-      SetValid(false);
-      setErr("username must be unique");
-      setTimeout(() => {
-        SetShowSnackBar(false);
-      }, 1000);
-    }
-  }
+	const [user, setUser] = useState<User>(new User('', ''));
+	const [valid, SetValid] = useState<boolean>(true);
+	const [showSnackBar, SetShowSnackBar] = useState(false);
+	const [err, setErr] = useState('');
+	const navigate = useNavigate();
+	const [signIn] = useAuth();
+	const {checkUser} = useServiceLayer();
 
-  return (
-    <div className="loginpage">
-      <div className="maincontainer">
-        <h1>Login</h1>
-        <div className="form">
-          <Typography variant="h6">
-            Create Your Account {' '}  
-            
-            <span>
-              <Link to="/register">Register</Link>
-            </span>
-          </Typography>
-          <TextField
-            required
-            id="outlined-required"
-            label="Username"
-            onChange={(e) => {
-              setUser(new User(e.target.value, user.password!));
-            }}
-          />
-          <TextField
-            required
-            id="outlined-required"
-            label="Password"
-            type="password"
-            onChange={(e) => {
-              setUser(new User(user.name!, e.target.value));
-            }}
-          />
-          {isValid(user) ? (
-            <Button
-              size="large"
-              variant="contained"
-              className="loginbutton"
-              onClick={validateUser}
-            >
+
+	function isValid(user: User) {
+		if (user!.password!.length! >= 8 && user!.name!.trim().length > 1) {
+			return true;
+		}
+		return false;
```

Comment:
update to 

return `user!.password!.length! >= 8 && user!.name!.trim().length > 1`